### ISSUE TITLE
import&export error(s)

### ISSUE THREAD

**Issue by ilippert** [ccbogel/QualCoder#827]
**Describe the bug**
I am trying to import files, and get errors

**To Reproduce**
I tried to import and export mp4 and pdf files; and neither worked.

**Expected behavior**
no error messages.

**Desktop (please complete the following information):**
 - OS: Fedora Linux 
 - Version 38

**Additional context**
I get these errors:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/Qualcoder-3.4-py3.11.egg/qualcoder/manage_files.py", line 1365, in import_files
    copyfile(f, destination)

  File "/usr/lib64/python3.11/shutil.py", line 258, in copyfile
    with open(dst, 'wb') as fdst:
         ^^^^^^^^^^^^^^^

FileNotFoundError: [Errno 2] No such file or directory: '/home/il/Documents/Studies/Jobs/Current Jobs/Frankfurt/Hessencloud/Teaching/deep.teaching/QualCoder project/deep.teaching.qda/video/beyond.mp4
```

and also 
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/Qualcoder-3.4-py3.11.egg/qualcoder/manage_files.py", line 1368, in import_files
    self.load_media_reference("video:" + link_path)

  File "/usr/local/lib/python3.11/site-packages/Qualcoder-3.4-py3.11.egg/qualcoder/manage_files.py", line 1439, in load_media_reference
    cur.execute("insert into source(name,memo,owner,date, mediapath, fulltext) values(?,?,?,?,?,?)",

OperationalError: attempt to write a readonly database
```

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/Qualcoder-3.4-py3.11.egg/qualcoder/manage_files.py", line 378, in table_menu
    self.import_linked_file(id_, mediapath)

  File "/usr/local/lib/python3.11/site-packages/Qualcoder-3.4-py3.11.egg/qualcoder/manage_files.py", line 649, in import_linked_file
    copyfile(mediapath[6:], self.app.project_path + "/video/" + filename)

  File "/usr/lib64/python3.11/shutil.py", line 258, in copyfile
    with open(dst, 'wb') as fdst:
         ^^^^^^^^^^^^^^^

FileNotFoundError: [Errno 2] No such file or directory: '/home/il/Documents/Studies/Jobs/Current Jobs/Frankfurt/Hessencloud/Teaching/deep.teaching/QualCoder project/deep.teaching.qda/video/Der.mp4'
```

upon exporting I also get an error
```
Cannot export as linked file
[Errno 2] No such file or directory: '/home/ilippert/Documents/QualCoder project/deep.teaching.qda/documents/Der Pakt-102.mp4.txt'
```

**Comment by ccbogel** [comment]
Hi. can you confirm, for example the export -that the file is actually there. Ps I am free to Zoom right now at is 9am here

**Comment by ccbogel** [comment]
Hi, Have you got read and write permissions for both the files being imported and exported?

https://stackoverflow.com/questions/3319112/sqlite-error-attempt-to-write-a-readonly-database-during-insert

**Comment by ilippert** [comment]
Yes, definitively, I have both read and write permissions

**Comment by ilippert** [comment]
But, no, the files are not successfully imported. The files are at the source location, but they never get copied into the QualCoder folder

**Comment by ilippert** [comment]
> Also, under manage > manage bad links to files (if this is enabled) to re-link to files where they have been moved.
`bad links to files` is disabled

**Comment by ilippert** [comment]
So, when I tried to import a few files the other day, only 1 time it worked; but the other times it did not work. I tried on two computers.

**Comment by ccbogel** [comment]
Perhaps there is some error message in the log file.
Are you free for a zoom chat. I cant suggest anything else for now.

**Comment by ccbogel** [comment]
OK, so  I tested the issue we discussed via zoom. That naming convention is not a problem. The issue was around not having a documents folder within the QualCoder project folder. So I do not know what happened there, maybe accidentally deleted? 
I have added an update that will create empty folder for documents, videos, audio, images if they are missing from the project folder,

**Comment by ccbogel** [comment]
Hi. 
Don't worry about ffmpeg not being present. (or that QualCoder cannot find the location of it) It would be looking for a standard system installation of it.
These below errors can be ignored, they are related to another contributor adding a lock to sqlite, will try and work to remove them:
'NoneType' object has no attribute 'stop'
remove: path should be string, bytes or os.PathLike, not NoneType`

Was the mp4 file opened at all? It yes, then copying the file cannot be performed, until the file is closed.
Secondly, **this is a guess**, maybe importing using OneDrive could be a problem? Perhaps try putting the file on the desktop and copy from there, similarity with the qualcoder project, put on your computer, but not in OneDrive.

**Comment by kaixxx** [comment]
> These below errors can be ignored, they are related to adding a lock to sqlite, will try and work to remove them:
> 'NoneType' object has no attribute 'stop'
> remove: path should be string, bytes or os.PathLike, not NoneType`

Fixed these two small errors quickly: https://github.com/ccbogel/QualCoder/pull/927. But they are most likely completely unrelated to the problems discussed here.

**Comment by lmbguy** [comment]
Thanks for the quick response!

While I was trying to import it the file was not open. I also tried importing the file from local storage which failed.
The import however did work after I relocated the qualcoder project to a local folder. Importing the mp4 from OneDrive was not a problem after I relocated the project.
For your reference, importing and working with .txt files while the project was located in Onedrive worked fine.

Thanks
